### PR TITLE
Electron flow

### PR DESCRIPTION
**Pull request by adartulloch**
PR to make electron flow speed in terms of Total Electricity Use gauge. I tried to evade hardcoding as best as possible, so it is actually more likely that we want the `animators.duration` of each powerline glyph to be somewhere around `5000` (which I tested and seemed to work the nicest for a noticeable visual effect) in the database.  

A list of the powerline glyph IDs: 
`5d320e770b4b1b2d8f42fdf0`
`5d320f490b4b1b2d8f42fdf1` 
`5d320d120b4b1b2d8f42fdef`
`5d321a600b4b1b2d8f42fdf4`
`5d3218e10b4b1b2d8f42fdf2`
`5d320ba80b4b1b2d8f42fdee`
`5d3219850b4b1b2d8f42fdf3`

Or this query should list them all in Robo 3T : `db.getCollection('activeGlyphs').find({'name' : 'powerline'})`

For each of these glyphs, we want to add this attribute at the top level:
`"data_url" : "https://qa.communityhub.cloud/data-hub/meters/compare/1414c3a0-cda7-11e8-9c46-02420aff0702/1414c3a0-cda7-11e8-9c46-02420aff0702"`

In the future, it might be feasible for `animation.duration` to be renamed `animation.maxDuration` for clarity.

### CODE REVIEW

File: cwd/src/factory.js [hunk 1]
```diff
@@ -131,11 +131,16 @@
             );
             break;
           case 'pathMover':
+            let animationDur = animator.duration;
+            if (glyph.name === 'powerline') {
+             let target = glyph.data;
```
[the remark this answers is not in the document]
Comment (Sammidysam) [reply cwd/src/factory.js:136]: I actually did get an error I think.

**Comment by Sammidysam** [comment]
https://drive.google.com/file/d/1N9b1YMAoogqjsEXjiK3RU0KzVOdEYM4w/view?usp=sharing
I wonder if we want to look further at the function we use to scale the duration?  I feel like having a value of 99 makes this too fast.  Maybe we want to scale it from 0.5 seconds to 2 seconds or something?

File: cwd/src/factory.js [hunk 2]
```diff
@@ -131,11 +131,16 @@
             );
             break;
           case 'pathMover':
+            let animationDur = animator.duration;
+            if (glyph.name === 'powerline') {
```
[the remark this answers is not in the document]
Comment (adartulloch) [reply cwd/src/factory.js:135]: Sure, what if we use 'pathMover.ratio' perhaps? And is the plan for this to be the actual ratio, or just an indicator that the we'll have to calculate a scaled ratio based on the data for a new `animation.duration` ?

**Comment by adartulloch** [comment]
indeed, at first I had a function that checked that if the values of `target` are between 90 and 100 because that makes the animation a bit too fast. I can implement that again.

**Comment by Sammidysam** [comment]
@adartulloch Did you make some changes or should I finish this up?

One idea I had this morning was that we could configure a `duration` and a `min_duration` - if `min_duration` exists, we will scale linearly between the two.  That makes it the user's fault if their animations look bad.  That just allows for more control

File: cwd/src/factory.js [hunk 3]
```diff
@@ -131,11 +131,16 @@
...
+             let target = glyph.data;
+             animationDur = ((100 - target) / 100) * animator.duration;
```
[the remark this answers is not in the document]
Comment (Sammidysam) [reply cwd/src/factory.js]: Resolved by forcing a minimum duration always.

File: cwd/src/factory.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sammidysam) [reply cwd/src/factory.js]: Resolved by checking for data

File: cwd/src/factory.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sammidysam) [reply cwd/src/factory.js]: So sorry that I missed this - I don't really know how.

Resolved by using the presence of `maxDuration` as opposed to `duration` as the signifier that we will be scaling if we have data as opposed to not.

**Pull request merged**